src/com/sap/piper/cm/ChangeManagement.groovy

```diff
@@ -177,25 +177,49 @@ public class ChangeManagement implements Serializable {
         String credentialsId,
         String cmclientOpts = '') {
 
-        def args = [
-                '-tID', transportRequestId,
-                "\"$filePath\""
-            ]
+        // 1.) if needed in the context of fiori deploy (for 'ui5 build' this is needed):
```

Comment:
Preparing the runtime environment is missing: `npm install --global @ui5/cli @sap/ux-ui5-tooling`

Reply:
We need to provide a config file like:

```
specVersion: '1.0'
metadata:
  name: openui5-sample-app
type: application
builder:
  customTasks:
  - name: deploy-to-abap
    afterTask: replaceVersion
    configuration:
      target:
        url: <ENDPOINT> 
        client: 001
        auth: basic
      credentials:
        username: env:ABAP_USER
        password: env:ABAP_PASSWORD
      app:
        name: NAME
        package: PACKAGE 
        transport: A5DK000075
      exclude:
      - .*\.test.js
      - internal.md
```

this file can be provide to `fiori deploy` via `-c`

Reply:
apparently we can provide from the command line:
```
/**
 * Supported CLI parameters
 */
const paramMap = {
    '-c': 'config',
    '-d': 'destination',
    '-u': 'url',
    '-t': 'transport'
};
```.

What about the other parameters, like e.g. package?

Reply:
> Preparing the runtime environment is missing:
> 
> `npm install --global @ui5/cli @sap/ux-ui5-tooling @ui5/logger @ui5/fs`
 
I thought we provide a docker image with this installed?

> With the current version we end up in confirmation dialog. That was not the case in earlier versions. --> needs to be changed since in in CI/CD scenarios there is nobody who hits `Y`

The ux tools team is working on it. For now, a version that does not need interaction is @sap/ux-ui5-tooling@1.0.10

Reply:
@o-liver: sorry I didn't see your comment earlier

> I thought we provide a docker image with this installed?

This would be - of course - possible. The advantages would be:
* less time for executing the step
* there are currently e.g. deprecation warnings wrt transitive dependencies in the log caused by installing the fiorr toolset. That does not look nice, but we can't do anything. With having an image these log messages would not apprear (of course we have that log messages and the outdated deps in the image in case we would provide an image.

The disadvantage is:
* we have to maintain the image.

Anyway I vote for an approach supporting both, a plain node image as starting point and also a preconfigured image. This can be achieved by making the deploy dependencies configurable. In case an empty list is provided we assume there is an already prepared image configured.

> [getting the deploy config file] The ux tools team is working on it.

There are already some properties which can be handed over to the deploy command as command line parameter. Maybe we will end up in a state where we can provide all parameter we have to set like this. With that it might happen that we don't need the config file at all. Anyway I'm open for using another tool for creating the config file in case we need it - as long as it does not work in an interactive mode.